segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -568,31 +568,22 @@ public void initialize() throws Exception {
                         val snapshotFileName = NameUtils.getSystemJournalSnapshotFileName(containerId, snapshotInfo.getEpoch(), snapshotInfo.getSnapshotId());
                         log.debug("SystemJournal[{}] Snapshot info read. {} pointing to {}", containerId, snapshotInfo, snapshotFileName);
 
-                        // Step 2: Validate.
-                        return checkSnapshotExists(snapshotFileName)
-                                // Step 3: Read contents.
-                                .thenComposeAsync(v -> getContents(snapshotFileName), executor)
-                                // Step 4: Deserialize and return.
-                                .thenApplyAsync(contents -> readSnapshotRecord(snapshotInfo, contents), executor);
+                        // Step 2: Validate. and Read contents.
+                        return getContents(snapshotFileName, false)
+                                // Step 3: Deserialize and return.
+                                .thenApplyAsync(contents -> readSnapshotRecord(snapshotInfo, contents), executor)
+                                .exceptionally( e -> {
+                                    throw new CompletionException(new IllegalStateException(
```

Comment:
Whatever exception may come from `readSnapshotRecord` will be caught by this block and will be converted into this `IllegalStateException`, even though the problem may be unrelated to the check of chunk existence. Is this conceptually correct? Can this cause any problem downstream?

Reply:
IllegalStateException is the correct behavior. In ability to read already created and validated snapshot is a critical error that prevents SLTS from boot.
readSnapshotRecord we retry several times before throwing exception.

Reply:
However, the message certainly needs to be more generic

Reply:
fixed.